src/tensorlake/builder/client_v2.py

```diff
@@ -195,8 +196,25 @@ async def build(self, context: BuildContext, image: Image) -> BuildInfo:
         res.raise_for_status()
         build = BuildInfo.model_validate(res.json())
 
+        click.secho(f"Starting build for image {image.image_name} ...", fg="green")
         click.secho(f"Build ID: {build.id}", fg="green")
-        return await self.stream_logs(build)
+        try:
+            return await self.stream_logs(build)
+        except (asyncio.CancelledError, KeyboardInterrupt, click.Abort):
+            try:
+                click.secho(
+                    f"Cancelling build for image {image.image_name} ...", fg="yellow"
+                )
+                await self._cancel_build(build, image)
+                click.secho(
+                    f"Cancelled build for image {image.image_name}", fg="yellow"
+                )
+            except Exception as e:
+                click.secho(
+                    f"Failed to cancel build for image {image.image_name}: {e}",
+                    fg="red",
+                )
+            raise
```

Comment:
Why do we need to re-raise? Would this result in a clean error message to customer?